cudamapper/src/overlapper_triggered.cu

```diff
@@ -414,7 +420,7 @@ void OverlapperTriggered::get_overlaps(std::vector<Overlap>& fused_overlaps,
 
     device_buffer<Overlap> d_filtered_overlaps(n_fused_overlap, _allocator, _cuda_stream);
 
-    FilterOverlapOp filterOp(min_residues, min_overlap_len, min_bases_per_residue, min_overlap_fraction);
+    FilterOverlapOp filterOp(min_residues, min_overlap_len, min_bases_per_residue, min_overlap_fraction, query_index == target_index);
```

Comment:
@mimaric @vellamike The reason this PR is failing tests is because this condition can be triggered if both pointer-to-indexes are the same int (e.g., 0). Do you have any recommendations for a safer why to check equivalence of two indexes?

Reply:
This was fixed in the tests by passing `nullptr` in place of both indexes, but would still appreciate input on how to do this better.

Reply:
see my other comment, I think we can use `ApplicationParameters::all_to_all` (i.e. pass that bool to overlapper, no the whole `ApplicaitonParameters`)